models/comments-model.js

```diff
@@ -1,17 +1,59 @@
 const db = require("../db/connection.js");
 
-const fetcharticlesIds = () => {
+exports.fetcharticlesIds = () => {
   return db.query("Select article_id FROM articles;");
 };
 
-const fetchCommentsByArticleId = (id) => {
+exports.fetchCommentsByArticleId = (id) => {
   return db.query(
     "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
     [id]
   );
 };
 
-module.exports = {
-  fetchCommentsByArticleId,
-  fetcharticlesIds,
+exports.insertCommentOnArticle = async (id, username, body) => {
+  const articleIdIsValid = await checkArticleIdIsValid(id);
+  const usernameIsValid = await checkUsernameIsValid(username);
+
+  if (!usernameIsValid) {
```

Comment:
We should not need to do this for username or article ID as it should automatically throw a PSQL error  if not valid- we can set up an error handler file that can deal with such errors.

Reply:
Done